components/provisioner/internal/provisioning/service_test.go

```diff
@@ -626,6 +628,135 @@ func TestService_UpgradeRuntime(t *testing.T) {
+	cluster := model.Cluster{
+		ID: runtimeID,
+		ClusterConfig: model.GardenerConfig{
+			ClusterID:              runtimeID,
+			Purpose:                util.StringPtr("evaluation"),
+			LicenceType:            util.StringPtr("license"),
+			GardenerProviderConfig: providerConfig,
+		},
+	}
+
+	upgradeShootInput := newUpgradeShootInput("testing")
+	upgradedConfig, err := inputConverter.UpgradeShootInputToGardenerConfig(*upgradeShootInput.GardenerConfig, cluster.ClusterConfig)
+	require.NoError(t, err)
+
+	operation := model.Operation{
+		ClusterID: runtimeID,
+		State:     model.InProgress,
+		Type:      model.UpgradeShoot,
+		Stage:     model.WaitingForShootNewVersion,
+	}
+
+	operationMatcher := getOperationMatcher(operation)
+
+	t.Run("Should start runtime provisioning of Gardener cluster and return operation ID", func(t *testing.T) {
+		//given
+		sessionFactoryMock := &sessionMocks.Factory{}
+		readWriteSession := &sessionMocks.ReadWriteSession{}
+		writeSession := &sessionMocks.WriteSessionWithinTransaction{}
+		upgradeShootQueue := &mocks.OperationQueue{}
+		provisioner := &mocks2.Provisioner{}
+
+		sessionFactoryMock.On("NewReadWriteSession").Return(readWriteSession, nil)
+		readWriteSession.On("GetLastOperation", runtimeID).Return(lastOperation, nil)
+		readWriteSession.On("GetCluster", runtimeID).Return(cluster, nil)
+		sessionFactoryMock.On("NewSessionWithinTransaction").Return(writeSession, nil)
+		writeSession.On("UpdateGardenerClusterConfig", upgradedConfig).Return(nil)
+		provisioner.On("UpgradeCluster", runtimeID, upgradedConfig).Return(nil)
+		provisioner.On("setOperationStarted", writeSession, runtimeID, model.UpgradeShoot, model.WaitingForShootNewVersion, nil, nil)
+		writeSession.On("Commit").Return(nil)
+		writeSession.On("RollbackUnlessCommitted").Return()
+		writeSession.On("InsertOperation", mock.MatchedBy(operationMatcher)).Return(nil)
+		upgradeShootQueue.On("Add", mock.AnythingOfType("string")).Return(nil)
+
+		service := NewProvisioningService(inputConverter, graphQLConverter, nil, sessionFactoryMock, provisioner, uuidGenerator, nil, nil, nil, upgradeShootQueue)
+
+		//when
+		operationStatus, err := service.UpgradeGardenerShoot(runtimeID, upgradeShootInput)
+		require.NoError(t, err)
+
+		//then
+		assert.Equal(t, runtimeID, *operationStatus.RuntimeID)
+		assert.NotEmpty(t, operationStatus.ID)
+		sessionFactoryMock.AssertExpectations(t)
+		readWriteSession.AssertExpectations(t)
+		writeSession.AssertExpectations(t)
+		upgradeShootQueue.AssertExpectations(t)
+	})
+
+	for _, testCase := range []struct {
+		description string
+		mockFunc    func(sessionFactory *sessionMocks.Factory, readWriteSession *sessionMocks.ReadWriteSession, writeSession *sessionMocks.WriteSessionWithinTransaction, provisioner *mocks2.Provisioner)
+	}{
+		{
+			description: "should fail to upgrade Shoot when failed to commit transaction",
```

Comment:
Test case for failed shoot update is missing.

Reply:
To be sure. We mean failed DB shoot config update?

Reply:
> To be sure. We mean failed DB shoot config update?

Ok, after second look it seems the testcase I was thinking about is there but the name is a bit confusing: "should fail to upgrade Shoot when failed to upgrade cluster". "Should fail when failed to update shoot"?